tests/src/test/java/alluxio/testutils/IntegrationTestUtils.java

```diff
@@ -117,18 +195,24 @@ public static void waitForBlocksToBeFreed(final BlockWorker bw, final Long... bl
       throws TimeoutException {
     try {
       // Execute 1st heartbeat from worker.
-      HeartbeatScheduler.execute(HeartbeatContext.WORKER_BLOCK_SYNC);
+      Whitebox.getInternalState(bw, BlockMasterSync.class).heartbeat();
```

Comment:
why do we have to whitebox these?

Reply:
Because the heartbeat threads have had their time interval reduced, calling `HeartbeatScheduler.execute` causes almost guaranteed deadlock.

Reply:
hrmmm, how does that work? I thought if we manually scheduled heartbeats, the time intervals don't matter? We aren't using `ManuallyScheduleHeartbeat`?

Reply:
The test where this caused failures was using a manual heartbeat, but I think there may have been other heartbeats other than the BlockMasterSync which took resources and starved/deadlocked this heartbeat thread.

I couldn't figure out the source of the deadlock but the program would hang indefinitely without this change...

Reply:
The deadlock occurs in `FileOutStreamAsyncWriteJobIntegrationTest#renameScheduleFree`